Dockerfile

```diff
@@ -4,20 +4,52 @@ ENV TERM=xterm
 ENV JULIA_PROJECT=/opt/ice-floe-tracker-pipeline
 ENV JULIA_DEPOT_PATH=/opt/julia
 ENV JULIA_PKGDIR=/opt/julia
-
-RUN apt-get -y update && \
-    apt-get install -y git python3.10 && \
-    rm -rf /var/lib/apt/list/* 
+ENV JULIA_BUILD_PYCALL='ENV["PYTHON"]=""; using Pkg; Pkg.build()'
+ENV EBSEG_REPO='https://github.com/WilhelmusLab/ebseg.git'
+ENV IFTPIPELINE_REPO='https://github.com/WilhelmusLab/ice-floe-tracker-pipeline.git'
+ENV LOCAL_PATH_TO_IFT_CLI='/usr/local/bin/ice-floe-tracker.jl'
+
+# DEPENDENCIES
+#===========================================
+RUN apt-get update -y && \
+    apt-get -qq install -y \
+    build-essential \
+    cmake \
+    git \
+    wget \
+    unzip \
+    yasm \
+    pkg-config \
+    libswscale-dev \
+    libtbb-dev \
+    libjpeg-dev \
+    libpng-dev \
+    libtiff-dev \
+    libopenjp2-7-dev \
+    libavformat-dev \
+    libpq-dev \
+    python3.11 python3-pip
+
+# Python packages
+#===========================================
+RUN pip install --upgrade pip --break-system-packages
+
+RUN pip install git+${EBSEG_REPO} \
+    jinja2==3.1 \
```

Comment:
Do we need rasterio, numpy, and pandas here in this pip install?

Reply:
I think they are needed. I was trying to run things from a container and the core package wasn't building without those deps installed.

Reply:
In this Dockerfile, are numpy, pandas, and rasterio getting installed as deps or they just come with IFTpipeline in the Conda? I gues it's unclear why only some of the deps need to be pip installed in this Dockerfile, and others do not.

Reply:
I just realized that this is the Julia Dockerfile. We have the Python Dockerfile that already installs gdal/proj, requests, and skyfield. Maybe it is better to update the python to 3.11 there and install the python things there? We had to do this because mixing Python deps with Julia's Conda caused downstream problems when using Cylc.

see [this Dockerfile](https://github.com/WilhelmusLab/ice-floe-tracker-pipeline/blob/main/workflow/scripts/Dockerfile)

Reply:
I see. I forgot about that other dockerfile. So you mean install fsd in this other image?

Reply:
I think that could work, would you be able to call from the Docker image in the IFTPipeline CLI though? It will definitely work for using Cylc. I guess an alternate would be to install in the IFT.jl Conda and add a module and then the pipeline CLI will have access to it?

Reply:
> I think that could work, would you be able to call from the Docker image in the IFTPipeline CLI though?
I don't think so. fsd would need to be installed in the same container as IFTPipeline for that to happen. That's why I am trying to add fsd to the IFTPipeline image.

Reply:
> It will definitely work for using Cylc.

To facilitate things for Cylc, jobs that require both fsd and IFT tasks will have to be run separately from their respective containers. That's why I am advocating for separate (Cylc) workflows for each.

Reply:
I would say try it, but we'll have to do an end-end test with Cylc to make sure everything works with the containers and Cylc, etc. We can always refactor if needed based on any errors, or it just works!

Reply:
So try putting fsd and IFTPipeline in the same image?

Reply:
> > It will definitely work for using Cylc.
> 
> To facilitate things for Cylc, jobs that require both fsd and IFT tasks will have to be run separately from their respective containers. That's why I am advocating for separate (Cylc) workflows for each.

Each task can also run from a separate container, it doesn't have to be a new workflow.

Reply:
Right. By a workflow I mean a separate Cycl file to create and run jobs.

Reply:
> In this Dockerfile, are numpy, pandas, and rasterio getting installed as deps or they just come with IFTpipeline in the Conda? I gues it's unclear why only some of the deps need to be pip installed in this Dockerfile, and others do not.

I think that numpy, pandas, and rasterio  are available from Conda but the others aren't?

Reply:
> Right. By a workflow I mean a separate Cycl file to create and run jobs.

It might not need a new workflow even, just a param that triggers the correct preprocessing template for that one Cylc task in the existing workflow.

Reply:
> It might not need a new workflow even, just a param that triggers the correct preprocessing template for that one Cylc task in the existing workflow.

There is always a possibility! Maybe in a separate CLI? It could be challenging, though, to create a single template that's versatile enough for various workflows.